File: SMP_Campaigns/src/main/java/com/sessionm/smp_campaigns/DeepLinkActivity.java
package com.sessionm.smp_campaigns;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

public class DeepLinkActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_deep_link);

        String url = getIntent().getStringExtra("url");

        TextView textView = (TextView) findViewById(R.id.deep_link_textview);
        textView.setText(url);
    }
}

File: SMP_Kotlin/src/main/java/com/sessionm/smp_kotlin/DeepLinkActivity.kt
package com.sessionm.smp_kotlin

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.TextView

class DeepLinkActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_deep_link)

        val url = intent.getStringExtra("url")

        val textView = findViewById<TextView>(R.id.deep_link_textview) as TextView
        textView.text = url
    }
}
